### PR TITLE
Add yaml

### PR DESCRIPTION
**Pull request by DeliciousBounty**
Hello,
This is my PR in response to the #111 issue.
Now  YAML files are supported

### CODE REVIEW

File: cherrybomb-engine/src/lib.rs [hunk 1]
```diff
@@ -24,27 +24,43 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
...
-    verbose_print(config, None, "Parsing OAS file...");
-    let oas_json: Value = match serde_json::from_str(&oas_file) {
-        Ok(json) => json,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error parsing OAS file: {}", e));
+    // Reading OAS file to string let oas_json: Value;
+
+    let oas_json: Value;
+    let oas: OAS3_1;
+    match config.file.extension() {
```
Comment (GuyL99) [inline cherrybomb-engine/src/lib.rs:32]: This should be done with a let oas_json = match, not by declaring a mut variable in the function scope

File: cherrybomb-engine/Cargo.toml [hunk 2]
```diff
@@ -13,12 +13,13 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-cherrybomb-oas = "^0.1"
+cherrybomb-oas = {path = "../cherrybomb-oas"}
```
Comment (GuyL99) [inline cherrybomb-engine/Cargo.toml:16]: There were no changes to cherrybomb-oas, why did you even change it to path?

File: Cargo.toml [hunk 3]
```diff
@@ -24,7 +24,7 @@ clap = { version = "4.0.29", features = ["derive"] }
 const_format = "0.2.30"
 serde = { version = "^1.0", features = ["derive"] }
 serde_json = "^1.0"
-cherrybomb-engine = "^0.1"
+cherrybomb-engine = {path = "cherrybomb-engine"}
```
Comment (GuyL99) [inline Cargo.toml:27]: We need to uplaod the engine to crates.io and then change it to 0.1 again...

File: cherrybomb-engine/src/lib.rs [hunk 4]
```diff
@@ -24,27 +24,43 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
...
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    oas_json = match serde_json::from_str(&oas_file) {
+                        Ok(json) => json,
+                        Err(e) => return Err(anyhow::anyhow!("Error parsing OAS file: {}", e))?,
+                    };
+                }
+                Some("yaml") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    oas_json = match serde_yaml::from_str(&oas_file) {
```
Comment (GuyL99) [inline cherrybomb-engine/src/lib.rs:49]: Are you sure that serde_yaml::Value and serde_json::Value are the same for the compiler, I ran into trouble with this in the past, but I think there is a converter between them or something

File: cherrybomb-engine/src/lib.rs [hunk 5]
```diff
@@ -60,7 +76,7 @@ pub async fn run(config: &Config) -> anyhow::Result<Value> {
     }
 }
 
-fn run_profile_info(config: &Config, oas: &OAS3_1, oas_json: &Value) -> anyhow::Result<Value> {
+fn run_profile_info(config: &Config, _oas: &OAS3_1, oas_json: &Value) -> anyhow::Result<Value> {
```
Comment (GuyL99) [inline cherrybomb-engine/src/lib.rs:79]: Why is the oas suddenly unnecessary?

**Review by GuyL99 — CHANGES_REQUESTED** [review]

File: cherrybomb-engine/src/lib.rs (hunk 5 above)
Comment (DeliciousBounty) [reply cherrybomb-engine/src/lib.rs:79]: no need of oas parameter  in the info profile , so I marked it unnecessary

File: cherrybomb-engine/src/lib.rs (hunk 4 above)
Comment (DeliciousBounty) [reply cherrybomb-engine/src/lib.rs:49]: What do you mean the same for the compiler?  I used serde_yaml for the yaml struct and serde_json for the json struct.
They are completely separate types and have different definition.

File: cherrybomb-engine/src/lib.rs (hunk 5 above)
Comment (GuyL99) [reply cherrybomb-engine/src/lib.rs:79]: ok

File: cherrybomb-engine/src/lib.rs (hunk 1 above)
Comment (DeliciousBounty) [reply cherrybomb-engine/src/lib.rs:28]: rewrote the function :) 
Should be good now

File: cherrybomb-engine/src/lib.rs [hunk 6]
```diff
@@ -23,44 +23,51 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
...
+            };
+            let oas_json: Value = match ext.to_str() {
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
+                    match serde_json::from_str(&oas_file) {
+                        Ok(json) => json,
+                        Err(e) => return Err(anyhow::Error::msg(format!("Error parsing OAS file: {}", e))),
+                    }
+                }
+                Some("yaml") => {
```
Comment (jayvdb) [inline cherrybomb-engine/src/lib.rs:43]: this should also support `yml`.
However extension isnt really a good way to do this.
trying json, and falling back to yaml is another approach. c.f. https://github.com/oxidecomputer/progenitor/pull/227
my preferred approach is to look at the first non-whitespace char to see if it is json, e.g. https://github.com/paperclip-rs/paperclip/blob/43cfea1/src/v2/mod.rs#L110

File: cherrybomb-engine/src/lib.rs [hunk 7]
```diff
@@ -23,44 +23,51 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
...
+    match config.file.extension() {
+        Some(ext) => {
+            verbose_print(config, None, "Reading OAS file...");
+            let oas_file = match std::fs::read_to_string(&config.file) {
+                Ok(file) => file,
+                Err(e) => return Err(anyhow::Error::msg(format!("Error opening OAS file: {}", e))),
+            };
+            let oas_json: Value = match ext.to_str() {
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
```
Comment (jayvdb) [inline cherrybomb-engine/src/lib.rs:37]: "Parsing OAS json file..." to make better use of this stdout line.
likewise other cli output in this function.

**Comment by jayvdb** [comment]
@DeliciousBounty are you able to continue this? @GuyL99 , can you re-review?

File: cherrybomb-engine/src/lib.rs (hunk 7 above)
Comment (DeliciousBounty) [reply cherrybomb-engine/src/lib.rs:37]: What do you suggest?

File: cherrybomb-engine/src/lib.rs (hunk 6 above)
Comment (DeliciousBounty) [reply cherrybomb-engine/src/lib.rs:43]: @GuyL99 Yes serde_yaml::Value and serde_json::Value can be used here without any trouble . 
But which approach do you prefer by read the file itself or only read the extension ?
And then please approve my PR :)

**Review by GuyL99 — APPROVED** [review]

**Pull request merged**